### PR TITLE
feat: update the default configuration file name suffix

### PR DESCRIPTION
**Pull request by misitebao**
Change the suffix of the configuration file generated by initialization to `.yml`.

- [x] Update Code
- [x] Update Changelog

Resolve: https://github.com/go-task/task/issues/1061

### CODE REVIEW

**Review by pd93 — APPROVED** [review]
Makes sense to me seeing as we're using `.yml` in all our examples and internally.

File: docs/docs/changelog.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ sidebar_position: 7
 
 # Changelog
 
+## Unreleased
+
+### Changed
+
+- Change the name of the file generated by `task --init` from `Taskfile.yaml` to `Taskfile.yml`.
+
```
[the remark this answers is not in the document]
Comment (misitebao) [reply docs/docs/changelog.md]: Thanks for the review, it has been removed.

**Comment by andreynering** [comment]
Just a FYI that the change to `.yml` to `.yaml` done in the past was intentional: https://github.com/go-task/task/pull/621.

This is tricky, because in theory `.yaml` is the official recommendation, but `.yml` is way more popular in the wild.

This ends up being personal preference, and I don't doubt someone may open another PR in the future to change from `.yml` back to `.yaml`.

To be honest, I don't have a strong opinion. What do you think?

**Comment by misitebao** [comment]
@andreynering 
In addition to the reasons I mentioned in https://github.com/go-task/task/issues/1061#issue-1624863444, another reason is the `Taskfile.yml` used by the examples and [styleguides](https://taskfile.dev/styleguide/#use-taskfileyml-and-not-taskfileyml) in our documentation.

**Comment by pd93** [comment]
> To be honest, I don't have a strong opinion. What do you think?

It seems like we didn't have support for `.yaml` at all before #621? We should definitely support both extensions, but I don't have a strong opinion on the "defacto" one.

I'm a big fan of consistency though. So if we stick to `.yaml` in `--init`, I feel that this project should stick to it too). Maybe it's easier to conform to `.yml` at this point, since anyone copying from our docs/examples will have used that instead of `.yaml`.

**Comment by andreynering** [comment]
OK, sticking with `.yml` which is our internal convention.

**Pull request merged**